### ISSUE TITLE
Error Installing on Ubuntu 16.04 LTS

### ISSUE THREAD

**Issue by arobinsongit** [chrissanders/FlowBAT#20]
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04 LTS
Release:        16.04
Codename:       xenial

./install_flowbat_ubuntu.sh: line 141: npm: command not found
To manually start, stop, or check status of FlowBAT:
sudo service flowbat [start/stop/status]

Attempting startup. Check http://127.0.0.1:1800.
Failed to start flowbat.service: Unit flowbat.service not found.

Logfiles attached
[logfile.txt](https://github.com/chrissanders/FlowBAT/files/253706/logfile.txt)
[silkinstall.log.txt](https://github.com/chrissanders/FlowBAT/files/253707/silkinstall.log.txt)
[silkonabox.sh.txt](https://github.com/chrissanders/FlowBAT/files/253708/silkonabox.sh.txt)
[install_flowbat_ubuntu.sh.txt](https://github.com/chrissanders/FlowBAT/files/253705/install_flowbat_ubuntu.sh.txt)

**Comment by automayt** [comment]
This is due to a number of changes from ubuntu 14.04 to 16.04 including but not limited to the nodejs implementation and the change from upstart. Will try to get a new installer for 16.04.

**Comment by kmneill01** [comment]
Any progress on this issue? Having the same problem with install on latest version of Linux Mint

**Comment by automayt** [comment]
This issue should be fixed soon, but nothing yet. It will require a
different install process currently, so I'm working to streamline that.

**Comment by MrMxyzptlk** [comment]
Ha, I was all excited...almost done with the install...and then...same error as above.
I'm running Mint 18.1, I guess you have to create something for systemd?

Can I manually start it?

**Comment by automayt** [comment]
Yes, and a symlink for node->nodejs basically. Ubuntu16.04 compatibility is next on the update list. As far as manual starts go, you can either run "sudo service flowbat start" (on 14.04) or node /pathtoflowbatdirectoryhere/private/bundle/main.js. In 16.04 and above you might have to specify nodejs instead of just node.

**Comment by Dasutin** [comment]
I was able to get it working by using nvm to backport the version of node to 0.10.48.  

I also created a systemd service "flowbat.service" in /etc/systemd/system

```
[Unit]
Description = Starting FlowBAT
After = network.target
[Service]
ExecStart = node /pathtoflowbatdirectoryhere/private/bundle/main.js
[Install]
WantedBy = multi-user.target
```

Then enable the new service and start it:
systemctl enable flowbat.service
systemctl start flowbat.service

**Comment by automayt** [comment]
Thanks for the heads up!

**Comment by cdnvballer99** [comment]
Hi, 
do you have any working installer for Ubuntu 16.04.1 LTS ?
I just tried the installation for 1st time, and failed --  proceeded to try to get node.js installed, nvm,  and tried the last section posted by Dasutin -- 

output looks like this:
sudo service flowbat status
● flowbat.service - Starting FlowBAT
   Loaded: loaded (/etc/systemd/system/flowbat.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Fri 2017-03-03 14:06:27 PST; 11min ago
 Main PID: 12083 (code=exited, status=8)

Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at packages/accounts-base/accounts_common.js:141:1
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at /usr/local/FlowBAT/private/bundle/programs/server/packages/accounts-base.js:243:4
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at /usr/local/FlowBAT/private/bundle/programs/server/packages/accounts-base.js:1659:3
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at /usr/local/FlowBAT/private/bundle/programs/server/boot.js:222:10
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at Array.forEach (native)
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at Function._.each._.forEach (/usr/local/FlowBAT/private/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
Mar 03 14:06:27 ubuntu-netsrv02 node[12083]:     at /usr/local/FlowBAT/private/bundle/programs/server/boot.js:117:5
Mar 03 14:06:27 ubuntu-netsrv02 systemd[1]: flowbat.service: Main process exited, code=exited, status=8/n/a
Mar 03 14:06:27 ubuntu-netsrv02 systemd[1]: flowbat.service: Unit entered failed state.
Mar 03 14:06:27 ubuntu-netsrv02 systemd[1]: flowbat.service: Failed with result 'exit-code'.

**Comment by norman25** [comment]
hum not workin in ubuntu 16.04 lts. next try which ubuntu 14.04  lts and again not works.

root@ubuntu:/home# ./install_flowbat_ubuntu.sh
It looks like mongod is running, which could be indicative that you have a meteor app running.
Are you sure you want to continue the install of FlowBAT? [y/n] y

Checking installed packages...

Checking for nodejs...

Checking for meteor...
**./install_flowbat_ubuntu.sh: line 141: npm: command not found**
To manually start, stop, or check status of FlowBAT:
sudo service flowbat [start/stop/status]

Attempting startup. Check http://127.0.0.1:1800.
service flowbat status
flowbat stop/waiting

**Comment by Ondjultomte** [comment]
Any updates here ? I want to get flowbat up and running

**Comment by chrissanders** [comment]
None yet. This has been on the back burner, but Jason is working on this and we're hoping to have it resolved in the next couple of weeks.

**Comment by Ondjultomte** [comment]
Is there a high possibility that it will be ready  in oct.? Or is it more likely the end of the year?

**Comment by Dasutin** [comment]
I wrote up some instructions on what worked for me to get FlowBAT installed on Ubuntu 16.04 LTS.  This should technically work on later versions of Ubuntu.  If I have time I'll see about modifying the install script. 

https://gist.github.com/Dasutin/643764bd8069ebf7063c7a9bc7f87893

**Comment by FlickeringIRoam** [comment]
The pull request here: https://github.com/chrissanders/FlowBAT/pull/28/commits/478430e3f5b746b52ab436e65f55d425ff94fd01
to force meteor to a specific version also fixes the fibers issue on ubuntu 16.04.
